target/linux/ath79/generic/base-files/etc/board.d/02_network

```diff
@@ -656,6 +660,9 @@ ath79_setup_macs()
 	wd,mynet-wifi-rangeextender)
 		lan_mac=$(nvram get et0macaddr)
 		;;
+	xiaomi,aiot-ac2350)
+		lan_mac=$(mtd_get_mac_binary art 0x1002)
+		;;
```

Comment:
This looks odd. Please provide an overview of vendor MAC address like https://openwrt.org/docs/guide-developer/device-support-policies#mac_addresses ...

Reply:
it is odd... but that's how xiaomi did it here. lan mac address is the same as the wlan mac (2.4ghz iirc)... i agree it should be documented.

Reply:
Well, after all we will probably find out that this line can be dropped and/or replaced by wan_mac setup.
But to decide about this, we first need the table I asked for.

Reply:
ok here goes:

```
MAC addresses as verified by OEM firmware:

use   address   source
WAN 88:C3:97:14:B5:57 art 0x0 (label)
LAN   88:C3:97:85:8D:2D  art 0x1002
2g    (same as LAN)
5g    88:C3:97:85:8D:2C     art 0x5006
```

so... 2g is lan is 5g+1. but wan is unrelated to either of them.

@eisaev since i don't have access to an ac2350 any more, can you confirm that WAN is indeed the "label" mac?

Reply:
This is the vendor's way.
On stock firmware:
```
root@XiaoQiang:/# ip addr
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 16436 qdisc noqueue state UNKNOWN 
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
    inet 127.0.0.1/8 scope host lo
    inet6 ::1/128 scope host 
       valid_lft forever preferred_lft forever
2: ifb0: <BROADCAST,NOARP> mtu 1500 qdisc noop state DOWN qlen 32
    link/ether 8e:c1:8b:1a:77:b5 brd ff:ff:ff:ff:ff:ff
3: ifb1: <BROADCAST,NOARP> mtu 1500 qdisc noop state DOWN qlen 32
    link/ether 1a:a4:93:39:c3:79 brd ff:ff:ff:ff:ff:ff
4: eth0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc pfifo_fast state UP qlen 1000
    link/ether 88:c3:97:13:b5:c6 brd ff:ff:ff:ff:ff:ff
    inet6 fe80::8ac3:97ff:fe13:b5c6/64 scope link 
       valid_lft forever preferred_lft forever
5: tunl0: <NOARP> mtu 1480 qdisc noop state DOWN 
    link/ipip 0.0.0.0 brd 0.0.0.0
6: miireg: <> mtu 0 qdisc noop state DOWN 
    link/generic 
7: gre0: <NOARP> mtu 1476 qdisc noop state DOWN 
    link/gre 0.0.0.0 brd 0.0.0.0
8: wifi0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc pfifo_fast state UNKNOWN qlen 2699
    link/ieee802.11 88:c3:97:82:8e:7a brd ff:ff:ff:ff:ff:ff
9: wifi1: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc pfifo_fast state UNKNOWN qlen 2699
    link/ieee802.11 88:c3:97:82:8e:79 brd ff:ff:ff:ff:ff:ff
10: br-lan: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP 
    link/ether 88:c3:97:82:8e:7a brd ff:ff:ff:ff:ff:ff
    inet 192.168.31.1/24 brd 192.168.31.255 scope global br-lan
11: eth0.1@eth0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue master br-lan state UP 
    link/ether 88:c3:97:82:8e:7a brd ff:ff:ff:ff:ff:ff
12: wlan0: <BROADCAST,MULTICAST> mtu 1500 qdisc noop state DOWN qlen 1000
    link/ether 88:c3:97:82:8e:78 brd ff:ff:ff:ff:ff:ff
13: p2p0: <BROADCAST,MULTICAST> mtu 1500 qdisc noop state DOWN qlen 1000
    link/ether 8e:c3:97:82:8e:78 brd ff:ff:ff:ff:ff:ff
14: eth0.2@eth0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP 
    link/ether 88:c3:97:13:b5:c6 brd ff:ff:ff:ff:ff:ff
    inet 192.168.99.7/24 brd 192.168.99.255 scope global eth0.2
    inet6 fe80::8ac3:97ff:fe13:b5c6/64 scope link 
       valid_lft forever preferred_lft forever
15: wl1: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue master br-lan state UNKNOWN 
    link/ether 88:c3:97:82:8e:7a brd ff:ff:ff:ff:ff:ff
    inet6 fe80::8ac3:97ff:fe82:8e7a/64 scope link 
       valid_lft forever preferred_lft forever
16: wl0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue master br-lan state UNKNOWN 
    link/ether 88:c3:97:82:8e:79 brd ff:ff:ff:ff:ff:ff
    inet6 fe80::8ac3:97ff:fe82:8e79/64 scope link 
       valid_lft forever preferred_lft forever
```
On OpenWRT:
```
root@OpenWrt:/# ip addr
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN qlen 1000
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
    inet 127.0.0.1/8 scope host lo
       valid_lft forever preferred_lft forever
    inet6 ::1/128 scope host 
       valid_lft forever preferred_lft forever
2: eth0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc fq_codel state UP qlen 1000
    link/ether 88:c3:97:13:b5:c6 brd ff:ff:ff:ff:ff:ff
    inet6 fe80::8ac3:97ff:fe13:b5c6/64 scope link 
       valid_lft forever preferred_lft forever
4: wlan0: <BROADCAST,MULTICAST> mtu 1500 qdisc noop state DOWN qlen 1000
    link/ether 88:c3:97:82:8e:79 brd ff:ff:ff:ff:ff:ff
5: wlan1: <BROADCAST,MULTICAST> mtu 1500 qdisc noop state DOWN qlen 1000
    link/ether 88:c3:97:82:8e:7a brd ff:ff:ff:ff:ff:ff
6: br-lan: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP qlen 1000
    link/ether 88:c3:97:82:8e:7a brd ff:ff:ff:ff:ff:ff
    inet 192.168.1.1/24 brd 192.168.1.255 scope global br-lan
       valid_lft forever preferred_lft forever
    inet6 fd7e:fb92:48d0::1/60 scope global noprefixroute 
       valid_lft forever preferred_lft forever
    inet6 fe80::8ac3:97ff:fe82:8e7a/64 scope link 
       valid_lft forever preferred_lft forever
7: eth0.1@eth0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue master br-lan state UP qlen 1000
    link/ether 88:c3:97:82:8e:7a brd ff:ff:ff:ff:ff:ff
8: eth0.2@eth0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP qlen 1000
    link/ether 88:c3:97:13:b5:c6 brd ff:ff:ff:ff:ff:ff
    inet 192.168.99.7/24 brd 192.168.99.255 scope global eth0.2
       valid_lft forever preferred_lft forever
    inet6 fe80::8ac3:97ff:fe13:b5c6/64 scope link 
       valid_lft forever preferred_lft forever
```

Reply:
Strange setup, but based on this data you did it right in the PR.